tests/test_content_loader.py

```diff
@@ -2033,13 +2029,83 @@ def test_caching_of_messages(self, mock_read_yaml):
         mock_read_yaml.assert_called_once_with('content/frameworks/g-cloud-7/messages/index.yml')
 
     def test_load_message_raises(self, mock_read_yaml):
-
         mock_read_yaml.side_effect = IOError
         messages = ContentLoader('content/')
 
         with pytest.raises(ContentNotFoundError):
             messages.load_messages('not-a-framework', ['index'])
 
+    def test_get_metadata(self, mock_read_yaml):
+        mock_read_yaml.return_value = {
+            'field_one': 'value_one',
+            'field_two': 'value_two',
+        }
+        metadata = ContentLoader('content/')
+        metadata.load_metadata('g-cloud-7', ['index'])
+
+        assert metadata.get_metadata('g-cloud-7', 'index') == ContentMetadata({
+            'field_one': 'value_one',
+            'field_two': 'value_two',
+        })
+
+        assert metadata.get_metadata('g-cloud-7', 'index').field_one == 'value_one'
+
+    def test_get_metadata_with_key(self, mock_read_yaml):
+        mock_read_yaml.return_value = {
+            'field_one': 'value_one',
+            'field_two': 'value_two',
+        }
+        metadata = ContentLoader('content/')
+        metadata.load_metadata('g-cloud-7', ['index'])
+
+        assert metadata.get_metadata('g-cloud-7', 'index', 'field_one') == 'value_one'
+
```

Comment:
Coud test the error case of `get_metadata` when the block isn't found.

Reply:
Basically all of these tests are copied from `test_messages`, but yeah you're right on the whole that we could probably do more. I'll review.